src/h3/mod.rs

```diff
@@ -936,20 +978,42 @@ impl Connection {
     fn send_grease_frames(
         &mut self, conn: &mut super::Connection, stream_id: u64,
     ) -> Result<()> {
-        let mut d = [42; 128];
-        let mut b = octets::Octets::with_slice(&mut d);
+        let mut d = [0; 8];
+
+        let stream_cap = conn.stream_capacity(stream_id)?;
+
+        let grease_frame1 = grease_value();
+        let grease_frame2 = grease_value();
+        let grease_payload = b"GREASE is the word";
+
+        let overhead = octets::varint_len(grease_frame1) + // frame type
+            1 + // payload len
+            octets::varint_len(grease_frame2) + // frame type
+            1 + // payload len
+            grease_payload.len(); // payload
+
+        // Don't send GREASE if there is not enough capacity for it.
+        if stream_cap < overhead {
+            return Ok(());
```

Comment:
the downside with returning Ok(() here is that the function returned to then immediately calls `self.frames_greased = true;` and will never try to grease again, so we'll miss greasing certain classes of client connection.

Reply:
Yeah, that was kind of on purpose, as a slightly more complicated handling of this doesn't seem worth it. I expanded the comment to make it more clear.